### DISCUSSION TITLE
Large scale art project featuring PID controlled DC motor, absolute rotatory encoder with RS485 Modbus RTU Serial

### DISCUSSION THREAD

**Discussion by Alive24** [micropython/micropython#9646] [Show and tell]
# Repo Link: https://github.com/Alive24/BigTape

## Highlights
- Running on three ESP32 modules with MicroPython, and communicate through HTTP on the local network.
    - See specs [`here`](https://item.taobao.com/item.htm?spm=a1z09.2.0.0.23412e8dIjpdVC&id=544098298735&_u=h5oppnr8c07)
- Async PID control based on position with [`MicroPython simple-pid`](https://micropython-simple-pid.readthedocs.io/en/latest/#)
- Absolute Rotatory Encoder through RS485 and ModBus RTU Serial with [`MicroPython Modbus library`](https://github.com/brainelectronics/micropython-modbus)
    - See specs of the absolute encoder [here](http://www.buruiter.com/col.jsp?id=116)
    - In the actual project, a LoRa Transmission Module is used for remote communication in RS485 and Modbus RTU Serial. See specs [here](https://item.taobao.com/item.htm?spm=a1z09.2.0.0.44c72e8dhUzPyZ&id=626911956873&_u=h5oppnrb721).
- DC Motors driven by H-Bridge with PWM.
    - See spects for the H-Bridge [here](https://item.taobao.com/item.htm?spm=a1z09.2.0.0.23412e8dIjpdVC&id=541299572681&_u=h5oppnr0e11)
- Dynamically setting PID position through HTTP server with [`MicroDot`](https://github.com/miguelgrinberg/microdot)
- Triggerable by human movement based on [`Xiaomi Human Body sensor`](https://detail.tmall.com/item.htm?spm=a230r.1.14.29.26cad854OZ6heK&id=678617599483&ns=1&abbucket=4)

Feel free to post if you have any question!

---
update: I uploaded a video here: https://www.youtube.com/watch?v=SyM3ul_wpzs

**Comment by jimmo** [comment]
Wow, that is really cool @Alive24 !! Thanks for sharing. I don't suppose there's a video anywhere?

**Reply by Alive24** [reply]
> Wow, that is really cool @Alive24 !! Thanks for sharing. I don't suppose there's a video anywhere?

Thanks!

I just uploaded one, and you should be able to view it here: https://www.youtube.com/watch?v=SyM3ul_wpzs
